### PR TITLE
ENH: Add fmu.realization.is_reference to schema

### LINKED ISSUE #826: ENH: Add reference flag to realization block
Ref https://github.com/equinor/sumo/issues/153 and also https://github.com/equinor/fmu-dataio/issues/774

To solve the use case of tagging individual realizations as "reference", a reference flag of some kind is needed in the metadata structure. This would primarily be set by Sumo, so not during data production.

My suggestion is `fmu.realization.is_reference`.

Opinions?

### PR DESCRIPTION
**Pull request by tnatt**
closes #826

### CODE REVIEW

File: src/fmu/dataio/_model/fields.py [hunk 1]
```diff
@@ -298,6 +298,9 @@ class Realization(BaseModel):
     """The universally unique identifier for this realization. It is a hash of
     ``fmu.case.uuid`` and ``fmu.iteration.uuid`` and ``fmu.realization.id``."""
 
+    is_reference: bool = Field(default=False)
```
Comment (perolavsvendsen) [inline src/fmu/dataio/_model/fields.py:301]: Will this default it to `False` for _all data objects_ produced? If that's the case, that was not the intention. It should only be used within the `Realization` object, i.e objects of class="realization". So not the realization _block_ of every data object. We may have confused each other here 😕. Or am I confusing myself right now?

@equinor-ruaj do you want to add to the confusion?

**Review by perolavsvendsen — CHANGES_REQUESTED** [review]
See comment. If this populates the `fmu.realization.is_reference` in _all_ data objects produced (of any class), and defaults it to `False`, this will probably create problems down the line - or we would have to iterate through all data objects in a realization and change all of them upon flagging a realization as "reference".

File: src/fmu/dataio/_model/fields.py (hunk 1 above)
Comment (tnatt) [reply src/fmu/dataio/_model/fields.py:301]: You are completely correct @perolavsvendsen .. Nice catch 👍  this is on me, I understood the intention but messed up the implementation! In my head the new Realization object did not share pydantic models with the data models.. but they do of course! Will make necessary changes 😄 

..the schema is starting to get complicated 🤔

**Comment by jcrivenaes** [comment]
I have still issues with the wording here. Tried to have a discussion under the issue here: https://github.com/equinor/sumo/issues/153. 

Key message: propose and off-line discussion @perolavsvendsen, @tnatt

**Comment by tnatt** [comment]
> I have still issues with the wording here. Tried to have a discussion under the issue here: [equinor/sumo#153](https://github.com/equinor/sumo/issues/153).
> 
> Key message: propose an off-line discussion @perolavsvendsen, @tnatt

Yes lets do it! I am e.g. available after lunch today 🙂

**Review by mferrera — APPROVED** [review]
Based on our discussion last week I believe this is good to go

**Pull request merged**